src/main/java/bigqueryestatespring/services/EstatesService.java

```diff
@@ -0,0 +1,291 @@
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static bigqueryestatespring.exceptionMessages.ExceptionMessage.*;
+import static bigqueryestatespring.services.PropertiesAttribute.*;
+
+public class EstatesService implements Service {
+    private static String PATH_TO_CREDENTIALS = System.getenv("PATH_TO_GCLOUD_CREDENTIALS");
+    // list of columns to create tree
+    private List<String> columnNames = Arrays.asList(OPERATION, PROPERTY_TYPE, COUNTRY_NAME, STATE_NAME);
+    // aggregate function to process the last element
+    private OperationType operationType = OperationType.AVERAGE;
+    // last element of the tree (will be processed in aggregate function)
+    private String aggregateColumn = PRICE;
+    private String alias = operationType.getStringValue().toLowerCase() + '_' + aggregateColumn;
+
+    public EstatesService() {}
+
+    /**
+     *
+     * @param columnNames list of columns to create tree (last element will be processed in aggregate function)
+     * @param operationType aggregate function to process the last element
+     */
+    public EstatesService(List<String> columnNames, OperationType operationType) {
+        this.columnNames = columnNames;
+        if (operationType != null) {
+            this.operationType = operationType;
+            this.alias = operationType.getStringValue().toLowerCase() + '_' + aggregateColumn;
+        }
+    }
+
+    /**
+     * Constructor with default operation type (AVG)
+     *
+     * @param columnNames list of columns to create tree (last element will be processed in aggregate function)
+     */
+    public EstatesService(List<String> columnNames) {
+        this.columnNames = columnNames;
+    }
+
+    /**
+     * Gets bigQuery instance with credentials for estates project
+     *
+     * @return bigQuery instance
+     */
+    private BigQuery getBigQueryInstance() {
+        try {
+            return BigQueryOptions.newBuilder()
+                    .setCredentials(
+                            ServiceAccountCredentials
+                                    .fromStream(new FileInputStream(PATH_TO_CREDENTIALS))
+                    ).build().getService();
+        } catch (FileNotFoundException ex) {
+            ex.printStackTrace();
+            throw new RuntimeException(CREDENTIALS_FILE_NOT_FOUND);
+        } catch (IOException ex) {
+            ex.printStackTrace();
+            throw new RuntimeException(EXCEPTION_WHILE_READING_CREDENTIALS);
+        }
+    }
+
+    /**
+     * Gets QueryJobConfiguration which contains sql query as
+     * SELECT columnNames(except last), operationType(columnNames(last))
+     * FROM *** WHERE surface >= bottom and surface <= top
+     * GROUP BY columnNames(except last)
+     *
+     * @param bottom low border for space
+     * @param top high border for space
+     * @return QueryJobConfiguration
+     */
+    private QueryJobConfiguration getQueryJobConfiguration(int bottom, int top) {
```

Comment:
is there any better way to generate such query? 
You can start with query template with substitutions (beware of SQL injection)
or use jook

```
// Fetch a SQL string from a jOOQ Query in order to manually execute it with another tool.
// For simplicity reasons, we're using the API to construct case-insensitive object references, here.
Query query = create.select(field("BOOK.TITLE"), field("AUTHOR.FIRST_NAME"), field("AUTHOR.LAST_NAME"))
                    .from(table("BOOK"))
                    .join(table("AUTHOR"))
                    .on(field("BOOK.AUTHOR_ID").eq(field("AUTHOR.ID")))
                    .where(field("BOOK.PUBLISHED_IN").eq(1948));
String sql = query.getSQL();
```